bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/table/sort/BootstrapOrderByBorder.java

```diff
@@ -0,0 +1,117 @@
+package de.agilecoders.wicket.core.markup.html.bootstrap.table.sort;
+
+
+import org.apache.wicket.extensions.markup.html.repeater.data.sort.ISortStateLocator;
+import org.apache.wicket.extensions.markup.html.repeater.data.sort.OrderByLink;
+import org.apache.wicket.extensions.markup.html.repeater.data.sort.SortOrder;
+import org.apache.wicket.markup.html.WebMarkupContainer;
+import org.apache.wicket.markup.html.border.Border;
+
+import de.agilecoders.wicket.core.markup.html.bootstrap.image.GlyphIconType;
+import de.agilecoders.wicket.core.markup.html.bootstrap.image.IconBehavior;
+import de.agilecoders.wicket.core.markup.html.bootstrap.image.IconType;
+
+/**
+ * A Bootstrap stylized component that wraps markup with an OrderByLink. This has the advantage of being able to add
+ * the attribute modifier to the wrapping element as opposed to the link, so that it can be attached
+ * to &lt;th&gt; or any other element.
+ * 
+ * A set of {@link IconType} must be provided.
+ * 
+ * For example:
+ * 
+ * &lt;th wicket:id="order-by-border"&gt;Heading&lt;/th&gt;
+ *
+ * @param <S> the type of the sorting parameter
+ * @author Eric Hamel <eric.hamel@me.com>
+ * 
+ */
+public abstract class BootstrapOrderByBorder<S> extends Border {
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * Construct.
+	 * 
+	 * @param id the component id
+	 * @param property the property to be sorted on
+	 * @param stateLocator the state locator
+	 */
+	public BootstrapOrderByBorder(final String id, final S property, final ISortStateLocator<S> stateLocator) {
+		this(id, property, stateLocator, new OrderByLink.DefaultCssProvider<S>());
+	}
+	
+	/**
+	 * Construct. 	
+	 * 
+	 * @param id the component id
+	 * @param property the property to be sorted on
+	 * @param stateLocator the state locator
+	 * @param cssProvider a custom css provider
+	 */
+	public BootstrapOrderByBorder(final String id, final S property, final ISortStateLocator<S> stateLocator, final OrderByLink.ICssProvider<S> cssProvider) {
+		super(id);
+
+		OrderByLink<S> link = newOrderByLink("orderByLink", property, stateLocator);
+		addToBorder(link);
+		add(new OrderByLink.CssModifier<S>(link, cssProvider));
+		link.add(getBodyContainer());
+		
+		WebMarkupContainer iconSort = new WebMarkupContainer("iconSort");
+		iconSort.setVisible(false);
+		link.add(iconSort);
+		
+		if(stateLocator.getSortState().getPropertySortOrder(property) == SortOrder.ASCENDING && ascendingIconType() != null){
```

Comment:
Extract `stateLocator.getSortState().getPropertySortOrder(property)` into a local variable and reuse it.